mlflow/tracking/default_experiment/databricks_notebook_experiment_provider.py

```diff
@@ -8,3 +17,33 @@ def in_context(self):
 
     def get_experiment_id(self):
         return databricks_utils.get_notebook_id()
+
+
+class DatabricksRepoNotebookExperimentProvider(DefaultExperimentProvider):
+    def in_context(self):
+        return databricks_utils.is_in_databricks_repo_notebook()
+
+    def get_experiment_id(self):
+        global _resolved_repo_notebook_experiment_id
+        if _resolved_repo_notebook_experiment_id:
+            return _resolved_repo_notebook_experiment_id
+
+        source_notebook_id = databricks_utils.get_notebook_id()
+        source_notebook_name = databricks_utils.get_notebook_path()
+        tags = {MLFLOW_EXPERIMENT_SOURCE_TYPE: "REPO_NOTEBOOK", MLFLOW_EXPERIMENT_SOURCE_ID: source_notebook_id}
+
+        # With the presence of the above tags, the following is a get or create in which it will
+        # return the corresponding experiment if one exists for the repo notebook.
+        # If no corresponding experiment exist, it will create a new one and return
+        # the newly created experiment ID.
+        try:
+            experiment_id = MlflowClient().create_experiment(source_notebook_name, None, tags)
+        except MlflowException as e:
+            if e.error_code == databricks_pb2.ErrorCode.Name(databricks_pb2.INVALID_PARAMETER_VALUE):
+                experiment_id = databricks_utils.get_notebook_id()
```

Comment:
Under what condition such an error is returned and should be treated with notebook ID instead? Could you clarify it via a comment?

Reply:
If repo notebook experiment creation isn't allowed for the workspace, we should fall back to the existing behavior. Will do